### ISSUE TITLE
Service 'Set Program' not working for Dishwasher

### ISSUE THREAD

**Issue by Keplernl** [astrandb/miele#85]
### The problem

The service Set Program won't work for my dishwasher. It does work via the Swagger and all other sensors are functioning in HA.

```
Logger: homeassistant.components.websocket_api.http.connection
Source: components/websocket_api/connection.py:141
Integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 3 april 2022 20:45:05 (4 occurrences)
Last logged: 20:55:55

[547503625840] Error handling message: Service set_program: 400 Bad Request
[547503625840] Error handling message: template value does not contain a dynamic template @ data['sequence'][0]['target']['device_id'][0]. Got None
[547696402544] Error handling message: Service set_program: 400 Bad Request
```

```
Logger: homeassistant.helpers.script.websocket_api_script
Source: helpers/script.py:1365
First occurred: 3 april 2022 20:45:05 (3 occurrences)
Last logged: 20:55:55

websocket_api script: Error executing script. Error for call_service at pos 1: Service set_program: 400 Bad Request


```
I use this the Set program service, select Dishwasher and use the following parameters (I am using version 0.19)

```
service: miele.set_program
data:
  programId: 6
target:
  device_id: 6488f286878bf7a43284991ad57369ff
```

Let me know if you need/want more information

### Version of miele integration having the issue?

0.19

### Version of Home Assistant Core having the issue?

2022.3.8

### Additional information

[config_entry-miele-2293fa67e57cb3d920e2a9f5911a7ffa.json.txt](https://github.com/astrandb/miele/files/8412394/config_entry-miele-2293fa67e57cb3d920e2a9f5911a7ffa.json.txt)

**Comment by astrandb** [comment]
It would be helpful if you could show the generated curl command from Swagger from a successful call. I don't need the line with Authorization token.

**Comment by Keplernl** [comment]
My apologies, I figured out the problems.

- The first thing that needs to be done, is that the device needs to be turned on before a program can be selected and started
- Secondly, apparently for my dishwasher programId number 6 doesn't work, which I used for testing within HA (and used a different number in swagger...)

So when I turn on the device via HA and then select program (not number 6!) via HA, it works perfectly! I will close this issue.
